### PR TITLE
Add DisableSyntax.noWhileLoops option

### LINKED ISSUE #900: Add DisableSyntax.noWhiles
When enabled Scalafix would report errors about usage of `while` loops.

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #900

### CODE REVIEW

**Comment by olafurpg** [comment]
Review @ssmarter

**Comment by olafurpg** [comment]
Review @smarter

File: scalafix-tests/input/src/main/scala/test/disableSyntax/DisableSyntaxBase.scala [hunk 1]
```diff
@@ -57,4 +58,8 @@ case object DisableSyntaxBase {
   override def finalize(): Unit = println("exit") // assert: DisableSyntax.noFinalize
 
   val Right(notFound) = 1.asInstanceOf[Either[String, String]]
+
+  while (true) () /* assert: DisableSyntax.while
+  ^^^^^
+  while loops should be avoided, consider using recursion instead */
```
Comment (gabro) [inline scalafix-tests/input/src/main/scala/test/disableSyntax/DisableSyntaxBase.scala]: can you add a test also for `do/while`?

Comment (olafurpg) [reply scalafix-tests/input/src/main/scala/test/disableSyntax/DisableSyntaxBase.scala]: Good idea! Done

**Pull request merged**